### PR TITLE
Update not found error message in vendor

### PR DESCRIPTION
**Pull request by linyguo**
Fixing #416 and #492

Before:

After:

### CODE REVIEW

File: api/pkg/apis/v1alpha1/vendors/federation-vendor.go [hunk 1]
```diff
@@ -254,9 +255,10 @@ func (f *FederationVendor) onRegistry(request v1alpha2.COARequest) v1alpha2.COAR
 		}
 		if err != nil {
 			if v1alpha2.IsNotFound(err) {
+				errorMsg := fmt.Sprintf("federation '%s' is not found", id)
```
Comment (msftcoderdjw) [inline api/pkg/apis/v1alpha1/vendors/federation-vendor.go:258]: site '%s' is not found.

**Comment by msftcoderdjw** [comment]
> Although the other vendors don't specifically return the 404, there still should be 404 cases. I think any get call and put calls with dependency resources should be checking the Not Found.
I agree. In other vendors, we even treat all error as InternalError. We need to polish them as well. You can create another PR for it if that is huge.

File: coa/pkg/apis/v1alpha2/types.go [hunk 2]
```diff
@@ -135,6 +136,29 @@ const (
...
+	case 404:
+		return NotFound
+	case 405:
+		return MethodNotAllowed
+	case 409:
+		return Conflict
+	case 500:
+		return InternalError
+	default:
+		return None
```
Comment (msftcoderdjw) [inline coa/pkg/apis/v1alpha2/types.go:158]: None is not that ideal. It doesn't carry any http status information. If httpstatus code is 4xx or 5xx, but not defined in COAError. I think it is still worthy to keep them. @linyguo , @iwangjintian what do you think?

File: api/pkg/apis/v1alpha1/utils/apiclient.go [hunk 3]
```diff
@@ -98,6 +98,28 @@ type (
...
+	)
+}
+
+func NewAPIError(state v1alpha2.State, msg string) APIError {
+	return APIError{
+		Code:    state,
+		Message: msg,
+	}
+}
+
```
Comment (RemindD) [inline api/pkg/apis/v1alpha1/utils/apiclient.go:122]: Could you add a helper function to convert API errors to Coa errors. It is useful when the service gets an API error from http server but need to return Coa error.

Comment (linyguo) [reply api/pkg/apis/v1alpha1/utils/apiclient.go:122]: updated.

**Review by msftcoderdjw — APPROVED** [review]

**Pull request merged**